src/App.js

```diff
@@ -45,74 +40,40 @@ const NoLogin = () => {
  * Main App component. Wrap the main react-router components together with
  * the various dialogs and error messages that may be needed.
  */
-class App extends React.Component {
-  constructor (props) {
-    super(props)
-    this.state = {
-      navigationConfirm: {
-        show: false,
-        confirmYes: undefined,
-        confirmNo: undefined,
-      },
-    }
+const App = ({ vms, visibility, config }) => {
+  if (!config.get('loginToken')) { // login is missing
+    return (
+      <ContainerFluid>
+        <NoLogin />
+      </ContainerFluid>
+    )
   }
 
-  resetUserConfirmation (show = false, confirmYes = undefined, confirmNo = undefined) {
-    this.setState({
-      navigationConfirm: { show, confirmYes, confirmNo },
-    })
+  let detailToRender = null
+  switch (visibility.get('dialogToShow')) {
```

Comment:
I think these bits can be cleaned up later.  I don't see how it gets activated or used at all.  I can strip all the dead code out, but I'm not sure that work should be part of this PR.